src/hotspot/cpu/riscv/macroAssembler_riscv.cpp

```diff
@@ -4667,13 +4667,19 @@ void MacroAssembler::rt_call(address dest, Register tmp) {
   }
 }
 
-void MacroAssembler::test_bit(Register Rd, Register Rs, uint32_t bit_pos, Register tmp) {
+void MacroAssembler::test_bit(Register Rd, Register Rs, uint32_t bit_pos) {
   assert(bit_pos < 64, "invalid bit range");
   if (UseZbs) {
     bexti(Rd, Rs, bit_pos);
     return;
   }
-  andi(Rd, Rs, 1UL << bit_pos, tmp);
+  int64_t imm = (int64_t)(1UL << bit_pos);
+  if (is_simm12(imm)) {
+    andi(Rd, Rs, imm);
```

Comment:
Since `imm` is guaranteed to be a signed 12-bit immediate in this block, we could call `and_imm12` directly instead of `andi`.

Reply:
Thanks for your review. fixed.